contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2323,13 +2326,38 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 
 							if (strcmp(defel->defname, "default_database") == 0)
 								login_options = lappend(login_options, defel);
+							else if (strcmp(defel->defname, "name_location") == 0)
+							{
+								int location = defel->location;
+								orig_loginname = extract_identifier(queryString + location);
+								login_options = lappend(login_options, defel);
+							}
+							else if (strcmp(defel->defname, "from_windows") == 0)
+							{
+								from_windows = true;
+								login_options = lappend(login_options, defel);
+							}
 						}
 
 						foreach(option, login_options)
 						{
 							stmt->options = list_delete_ptr(stmt->options,
 															lfirst(option));
 						}
+
+						if (orig_loginname)
+						{
+							login_options = lappend(login_options, 
+													makeDefElem("original_login_name",
+																(Node *) makeString(orig_loginname),
+																-1));
+						}
+
+						if (from_windows && orig_loginname)
```

Comment:
We should check the following scenario:
If from_windows is false, the input format can't be of format domain/login.

Reply:
Yes, we are going to address it in separate effort BABEL-3828 and BABEL-3844